### PR TITLE
Reworks exotic blood conversion handling so dead species can regain blood via transfusion

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Moves exotic blood conversion from metabolize, directly to injection, so species with exotic blood can regain it from IV bags / being fed from a glass when dead.

## Why It's Good For The Game

Species with exotic blood being unable to replenish blood is bad. Since species with blood die under 20% blood, if a slime is under 20% blood, they will not be able to regain any blood as they die from having under 20% blood.

With this setup, it works more like normal blood, handled when injected

## Changelog
:cl:
fix: Slimes can now be given slime blood from IV bags when dead to regain blood.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -577,12 +577,27 @@
...
 			if("OBJ")
 				R.reaction_obj(A, R.volume * volume_modifier)
 
+/datum/reagents/proc/handle_exotic_blood(atom/A, datum/reagent/R, method, volume_modifier)
+	if(!R || !method || method != REAGENT_INGEST)
+		return FALSE
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		if(R.id == H.dna.species.exotic_blood)
+			//this is the same formula from the species handle_reagent proc
```
Comment (hal9000PR) [inline code/modules/reagents/chemistry/holder.dm:594]: redundant comment refers to deleted code.

Comment (Qwertytoforty) [reply code/modules/reagents/chemistry/holder.dm]: removed

File: code/modules/reagents/chemistry/holder.dm [hunk 2]
```diff
@@ -577,12 +577,26 @@
...
 				R.reaction_obj(A, R.volume * volume_modifier)
 
+/datum/reagents/proc/handle_exotic_blood(atom/A, datum/reagent/R, method, volume_modifier)
+	if(!R || !method || method != REAGENT_INGEST)
+		return FALSE
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		if(R.id == H.dna.species.exotic_blood)
+			H.blood_volume = min(H.blood_volume + round(R.volume * volume_modifier, 0.1), BLOOD_VOLUME_NORMAL)
+			del_reagent(R)
```
Comment (hal9000PR) [inline code/modules/reagents/chemistry/holder.dm:595]:
```suggestion
			del_reagent(R.id)
```
otherwise it won't remove the reagent.

**Review by hal9000PR — APPROVED** [review]
LGTM

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**